pkg/controllers/csi/driver/volumes/host/publisher.go

```diff
@@ -31,120 +29,45 @@ import (
-	bindCfg, err := csivolumes.NewBindConfig(ctx, publisher.db, volumeCfg)
-	if err != nil {
-		return nil, err
-	}
-
-	if err := publisher.mountOneAgent(bindCfg.TenantUUID, volumeCfg); err != nil {
+func (pub *Publisher) PublishVolume(ctx context.Context, volumeCfg *csivolumes.VolumeConfig) (*csi.NodePublishVolumeResponse, error) {
+	if err := pub.mountOneAgent(volumeCfg); err != nil {
 		return nil, status.Error(codes.Internal, "failed to mount osagent volume: "+err.Error())
 	}
 
-	volume, err := publisher.db.GetOsAgentVolumeViaTenantUUID(ctx, bindCfg.TenantUUID)
-	if err != nil {
-		return nil, status.Error(codes.Internal, failedToGetOsAgentVolumePrefix+err.Error())
-	}
-
-	timestamp := time.Now()
-	if volume == nil {
-		storage := metadata.OsAgentVolume{
-			VolumeID:     volumeCfg.VolumeID,
-			TenantUUID:   bindCfg.TenantUUID,
-			Mounted:      true,
-			LastModified: &timestamp,
-		}
-		if err := publisher.db.InsertOsAgentVolume(ctx, &storage); err != nil {
-			return nil, status.Error(codes.Internal, fmt.Sprintf("failed to insert osagent volume info to database. info: %v err: %s", storage, err.Error()))
-		}
-	} else {
-		volume.VolumeID = volumeCfg.VolumeID
-		volume.Mounted = true
-		volume.LastModified = &timestamp
-
-		if err := publisher.db.UpdateOsAgentVolume(ctx, volume); err != nil {
-			return nil, status.Error(codes.Internal, fmt.Sprintf("failed to update osagent volume info to database. info: %v err: %s", volume, err.Error()))
-		}
-	}
-
 	return &csi.NodePublishVolumeResponse{}, nil
 }
 
-func (publisher *HostVolumePublisher) UnpublishVolume(ctx context.Context, volumeInfo *csivolumes.VolumeInfo) (*csi.NodeUnpublishVolumeResponse, error) {
-	volume, err := publisher.db.GetOsAgentVolumeViaVolumeID(ctx, volumeInfo.VolumeID)
-	if err != nil {
-		return nil, status.Error(codes.Internal, failedToGetOsAgentVolumePrefix+err.Error())
-	}
-
-	if volume == nil {
-		return &csi.NodeUnpublishVolumeResponse{}, nil
-	}
-
-	publisher.unmountOneAgent(volumeInfo.TargetPath)
-
-	timestamp := time.Now()
-	volume.Mounted = false
-	volume.LastModified = &timestamp
-
-	if err := publisher.db.UpdateOsAgentVolume(ctx, volume); err != nil {
-		return nil, status.Error(codes.Internal, fmt.Sprintf("failed to update osagent volume info to database. info: %v err: %s", volume, err.Error()))
-	}
-
-	log.Info("osagent volume has been unpublished", "targetPath", volumeInfo.TargetPath)
-
-	return &csi.NodeUnpublishVolumeResponse{}, nil
-}
-
-func (publisher *HostVolumePublisher) CanUnpublishVolume(ctx context.Context, volumeInfo *csivolumes.VolumeInfo) (bool, error) {
-	volume, err := publisher.db.GetOsAgentVolumeViaVolumeID(ctx, volumeInfo.VolumeID)
-	if err != nil {
-		return false, status.Error(codes.Internal, failedToGetOsAgentVolumePrefix+err.Error())
-	}
+func (pub *Publisher) mountOneAgent(volumeCfg *csivolumes.VolumeConfig) error {
```

Comment:
calling this `mountOneAgent` when all we do is provide a config volume isn't really correct

Reply:
[fixed](https://github.com/Dynatrace/dynatrace-operator/pull/4129/commits/3f347a0d75f68e9359fc5389a9ac28e9293c60dc)